subsys/spm/spm_internal.h

```diff
@@ -60,7 +61,7 @@ static inline u32_t num_spu_flash_regions(void)
 #else
 
 #define FLASH_SECURE_ATTRIBUTION_REGION_SIZE CONFIG_NRF_SPU_FLASH_REGION_SIZE
-#define NUM_FLASH_SECURE_ATTRIBUTION_REGIONS (DT_SOC_NV_FLASH_0_SIZE \
+#define NUM_FLASH_SECURE_ATTRIBUTION_REGIONS (FLASH_AREA_SIZE(storage) \
```

Comment:
This should be left as is. The change reads the size of the partition used by the settings subsystem. The DT symbol is a non configurable AFAICS. Also remove the include to flash_map.h